### ISSUE TITLE
depext: cygwin 32 now needs extra "I know what I am doing" option 

### ISSUE THREAD

**Issue by MSoegtropIMC** [ocaml/opam#5398]
Recently the cygwin team deprecated and archived 32 bit cygwin. In Coq Platform I stlll want to support it in the sense that one can build it if one desperately wants the 32 bit version. This might happen because Coq is all pointers, so the memory consumption of the 32 bit version is about half and one can run twice as many parallel jobs on the same machine compared to the 64 bit version. There are few users who care, but there are a few.

Coming to the point: calling cygwin setup for 32 bit cygwin now requires the extra option `--allow-unsupported-windows` to indicate one knows what one is doing. Is there any way to pass this option by depext?

**Comment by MSoegtropIMC** [comment]
P.S.: here is a pointer to a failing [CI log](https://github.com/coq/platform/actions/runs/3744145576/jobs/6364324809#step:5:1739)

**Comment by dra27** [comment]
I presume this is with cygwinports? (opam-repository-mingw)? For the Cygwin depext support for opam 2.2 (which @rjbou and I have been working on), the plan was not to support cygwin32 at all, given its retirement. That said, if it turns out it's just a matter of a switch, then perhaps we could!

**Comment by MSoegtropIMC** [comment]
Indeed, it is with `opam-repository-mingw` - I am happy to switch to something else anytime someone says it makes sense.

The only difference I have between 32 and 64 bit cygwin in my scripts is the name of the setup.exe file (it is different), and this option, so it should be painless to support it. I expect that usage will cease within a year if we say it is deprecated.

It might btw. also be an option to support 32 bit MinGW on 64 bit cygwin. The toolchain packages exist. I tried it a few years back and there the issue was that Windows appears to have some file system hack with the effect that certain DLL files are different depending on if the process opening the DLL is 32 or 64 bit. This had the effect that a linker targeting 32 bit but being itself compiled for 64 bit was seeing the wrong DLL. At least this was the status a few years (3?) back when I last tried it. One could get around this by using the 32 bit executable to copy the DLLs somewhere else and redirect the linker there. But it is really messy/hacky/... and I don't know if one can bootstrap this (how to link the 32 bit app used for copying). But it might be the better option since cygwin 32 is officially deprecated.
